contracts/AckToEarn.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "hardhat/console.sol";
+
+contract AckToEarn is Ownable, ReentrancyGuard {
+    using Counters for Counters.Counter;
+
+    event NewBid(address indexed from, address indexed to, uint amount);
+    event ReclaimBid(address indexed bidder, uint amount);
+
+    Counters.Counter private bidIds;
+
+    struct Bid {
+        uint id;
+        address bidder;
+        address recipient;
+        uint amount;
+        uint timestamp;
+        string message;
+        bool claimed;
+        bool exists;
+    }
+
+    mapping(address => mapping(uint => Bid)) private bidderBids;
+    mapping(address => mapping(uint => Bid)) private recipientBids;
+
+    Bid[] public bids;
+
+    uint constant bidExpiryThreshold = 7 days;
+
+    /**
+    * @notice Send a message and bid amount to a recipient
+    */
+    function sendBid(string memory message, address recipient) external payable {
+        require(recipient != address(0), "Recipient cannot be the zero address");
+
+        // The recipient gets 90% of the bid amount
+        uint recipientAmount = (msg.value * 90) / 100;
+        uint newBidId = bidIds.current();
+        Bid memory bid = Bid({
+            id: newBidId,
+            bidder: msg.sender,
+            recipient: recipient,
+            amount: msg.value,
+            message: message,
+            timestamp: block.timestamp,
+            claimed: false,
+            exists: true
+        });
+
+        bids.push(bid);
+
+        bidderBids[msg.sender][newBidId] = bid;
+        recipientBids[recipient][newBidId] = bid;
+
+        emit NewBid(msg.sender, recipient, msg.value);
+    }
+
+    /**
+    * @notice Returns all bids that have ever been made on the platform
+    */
+    function getBids() external view returns(Bid[] memory) {
+        return bids;
+    }
+
+    /**
+    * @notice Reclaims a specific set of expired bids that the bidder has made.
+    */
+    function reclaimBids(uint[] memory bidIdsToReclaim) external nonReentrant {
+        uint totalReclaimAmount = 0;
+
+        for (uint i = 0; i < bidIdsToReclaim.length; i++)
+        {
+            uint bidId = bidIdsToReclaim[i];
+            Bid memory bid = bidderBids[msg.sender][bidId];
+
+            if (!bid.exists) {
```

Comment:
where do we set `bid.exists` to `false`?

Reply:
We don't, the default value for a `bool` in Solidity is `false`: 

https://www.oreilly.com/library/view/solidity-programming-essentials/9781788831383/33b99fa0-47c7-4cc4-8f5b-2930580df9a8.xhtml#:~:text=The%20default%20value%20for%20bool%20in%20Solidity%20is%20false%20.

https://ethereum.stackexchange.com/a/52163